hamal:

  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  lifter_controller:
    type: velocity_controllers/JointTrajectoryController
    joints: 
      - lifter_joint
    #pid: {p: 10.0, i: 0.01, d: 1.0}
    constraints:
      goal_time: 0.0
      stopped_velocity_tolerance: 0.01
      lifter_joint:
        trajectory: 0.04
        goal: 0.02

    state_publish_rate: 25
    action_monitor_rate: 20
    stop_trajectory_duration: 0
    gains:
      lifter_joint: {p: 100,  d: 1, i: 1, i_clamp: 1}
    
  #diff_drive_controller:
  #  type: hamal_diff_drive_controller/HamalDiffDriveController
  #  
  #  left_wheel:
  #  
  #  right_wheel:
#
  #  publish_rate:
#
  #  velocity_rolling_window_size:
#
  #  wheel_radius:
#
  #  wheel_separation:
#
  #  use_position_for_odom:
#
  #  use_velocity_for_odom:
#
  #  odom_tf_enabled: false # Leave the tf publishing to EKF for now
#
  #  base_frame_id: base_link
#
  #  odom_frame_id: odom
#
  #  pose_covariance: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
#
  #  twist_covariance: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
#
  #  twist_command_timeout: 0.5
  #  
  #  twist_command_topic_name: "diff_drive_controller/odom"
#
  #  limiter:
  #    linear_x:
  #      has_velocity_limits: true
  #      has_accel_limits: true
  #      has_jerk_limits: false
#
  #      max_vel: 0.45
  #      min_vel: -0.45
#
  #      max_accel: 0.225
  #      min_accel: -0.225
#
  #      max_jerk: 1.0
  #      min_jerk: -1.0
#
  #    angular_z:
  #      has_velocity_limits: true
  #      has_accel_limits: true
  #      has_jerk_limits: false
#
  #      max_vel: 1.96
  #      min_vel: -1.96
#
  #      max_accel: 0.98
  #      min_accel: -0.98
#
  #      max_jerk: 3.92
  #      min_jerk: -3.92
#